src/app/components/List/EBookList.jsx

```diff
@@ -17,27 +19,84 @@ import PropTypes from 'prop-types';
  * @return {string}
  */
 const generateLink = (url, eReaderUrl, local, download, ebook) => {
-  const link = (local && ebook && !download) ? `${eReaderUrl}?url=${url}` : `${url}`;
-  const linkText = (ebook && !download) ? 'Read Online' : 'Download';
+  const link = local && ebook && !download ? `${eReaderUrl}?url=${url}` : `${url}`;
+  const linkText = ebook && !download ? 'Read Online' : 'Download';
   return <a href={`${link}`}>{`${linkText}`}</a>;
 };
 
+// generates an array of options for the dropdown
+const generateOption = (link, eReaderUrl) => {
+  const url = link.local && link.ebook && !link.download ? `${eReaderUrl}?url=${link.url}` : `${link.url}`;
+  const label = link.label;
+  let className;
+  if (!link.local) {
+    className = 'external';
+  }
+  return {
+    value: label,
+    url,
+    label,
```

Comment:
Are `url` and `label` used by the `Dropdown` component? It seems like it only reads the other values of this object

Reply:
it's kind of duplicated, but I thought it would be better for reading what this is doing in the future. url and label are used by the onChange, and the others are used by the dropdown